packages/shared-ux/modal/tabbed/src/context/index.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, {
+  createContext,
+  useContext,
+  useReducer,
+  useMemo,
+  useRef,
+  useCallback,
+  type PropsWithChildren,
+  type Dispatch,
+} from 'react';
+import { once } from 'lodash';
+
+interface IDispatchAction {
+  type: string;
+  payload: any;
+}
+
+export type IDispatchFunction = Dispatch<IDispatchAction>;
+
+export interface IMetaState {
+  selectedTabId: string | null;
+}
+
+type IReducer<S> = (state: S, action: IDispatchAction) => S;
+
+export interface ITabDeclaration<S = {}> {
+  id: string;
+  name: string;
+  initialState?: Partial<S>;
```

Comment:
Regarding this `initialState`, why do we build state management into the modal? What does it solve?